File: ShopTask/Models/ProductModel.cs
using ShopTask.DataAccess.Entities;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Web.Mvc;
using ShopTask.Resources;

namespace ShopTask.Models
{
    public class ProductModel
    {
        public int? Id { get; set; }

        [Required(ErrorMessageResourceType = typeof(Res), ErrorMessageResourceName = "EmptyProductTitle")]
        [Display(Name = "ProductModelTitle", ResourceType = typeof(Res))]
        public string Title { get; set; }

        [Display(Name = "ProductModelPrice", ResourceType = typeof(Res))]
        [RegularExpression(@"(^[1-9][0-9]*(\.[0-9]*)?)|(^0\.(([0-9]*)?[1-9]([0-9]*)?))$",
            ErrorMessageResourceType = typeof(Res),
            ErrorMessageResourceName = "IncorrectProductPrice")]
        public decimal Price { get; set; }

        [Display(Name = "ProductModelDescription", ResourceType = typeof(Res))]
        public string Description { get; set; }

        public int CategoryId { get; set; }

        public CategoryModel Category { get; set; }

        public SelectList Categories { get; set; }
    }
}

File: ShopTask/Frontend/app/products/models/product.model.ts
export class Product {
    private _id: number;
    private _title: string;
    private _price: number;
    private _description: string;
    private _category: any;

    public get id(): number {
        return this._id;
    }

    public get title(): string {
        return this._title;
    }

    public get price(): number {
        return this._price;
    }

    public get description(): string {
        return this._description;
    }

    public get category(): any {
        return this._category;
    }

    public set category(value : any) {
        this._category = value;
    }

    constructor(data: any) {
        if(data) {
            this._id = data.Id;
            this._title = data.Title;
            this._price = data.Price;
            this._description = data.Description;
            this._category = data.Category;
        }
     }

    public SetData(data: any){
        this._title = data.title;
        this._price = data.price;
        this._description = data.description;
        this._category = data.category;
    }
};